pkg/plugins/policies/meshloadbalancingstrategy/plugin/v1alpha1/plugin.go

```diff
@@ -94,40 +99,61 @@ func (p plugin) configureDPP(
 	// we configure clusters in a separate loop to avoid configuring the same cluster twice
 	for serviceName, conf := range serviceConfs {
 		if cluster, ok := clusters.Outbound[serviceName]; ok {
-			if err := p.configureCluster(cluster, conf.LoadBalancer); err != nil {
+			if err := p.configureCluster(cluster, conf); err != nil {
 				return err
 			}
+			if cluster.LoadAssignment != nil {
+				if err := ConfigureStaticEndpointsLocalityAware(proxy, endpoints, splitEndpoints, cluster, conf, serviceName); err != nil {
+					return err
+				}
+			}
 		}
 		for _, cluster := range clusters.OutboundSplit[serviceName] {
-			if err := p.configureCluster(cluster, conf.LoadBalancer); err != nil {
+			if err := p.configureCluster(cluster, conf); err != nil {
 				return err
 			}
+			if cluster.LoadAssignment != nil {
+				if err := ConfigureStaticEndpointsLocalityAware(proxy, endpoints, splitEndpoints, cluster, conf, serviceName); err != nil {
+					return err
+				}
+			}
+		}
+
+		if err := configureEndpoints(proxy, endpoints, splitEndpoints, serviceName, conf, rs); err != nil {
+			return err
 		}
-		configureEndpoints(proxy.Dataplane, endpoints, serviceName, conf)
 	}
 
 	return nil
 }
 
 func configureEndpoints(
-	dataplane *core_mesh.DataplaneResource,
+	proxy *core_xds.Proxy,
 	endpoints policies_xds.EndpointMap,
+	splitEndpoints policies_xds.EndpointMap,
 	serviceName string,
 	conf api.Conf,
-) {
-	var zone string
-	if inbounds := dataplane.Spec.GetNetworking().GetInbound(); len(inbounds) != 0 {
-		zone = inbounds[0].GetTags()[mesh_proto.ZoneTag]
+	rs *core_xds.ResourceSet,
+) error {
+	var localZone string
+	if inbounds := proxy.Dataplane.Spec.GetNetworking().GetInbound(); len(inbounds) != 0 {
+		localZone = inbounds[0].GetTags()[mesh_proto.ZoneTag]
+	}
```

Comment:
nit:
```
if tags := dp.Spec.TagSet().Values(mesh_proto.ZoneTag); len(tags) > 0 {
			localZone = tags[0]
		}
```
this handles more cases. If it's a provided gateway you don't have tags in inbound